### PR TITLE
Feature/#34

### LINKED ISSUE #34: Today's Special Component

### PR DESCRIPTION
**Pull request by jespernpedersen**

### CODE REVIEW

File: src/components/Dish/Dish.vue [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        color: #33363a;
+        position: relative;
+        display: inline-block;
+        border-radius: 12px;
+        width: 100%;
+        height: 15vw;
+    }
+    .dish-inner.vertical .dish-label {
+        display: inline-block;
+        padding-top: 15px;
```
Comment (DiogoGeada) [inline src/components/Dish/Dish.vue]: Converge padding top and padding bottom into one line; padding: 15px 0;

File: src/components/Dish/Dish.vue [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        padding-bottom: 15px;
+    }
+    img {
+        height: auto;
+        width: 100%;
+        position: absolute;
+        left: 50%;
+        top: 50%;
+        transform: translate(-50%, -50%);
+        border-radius: 12px;
```
Comment (DiogoGeada) [inline src/components/Dish/Dish.vue]: converge the 3 lines with border-radius: 12px 0 0 12px;

File: src/components/Dish/Dish.vue [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    .image-wrapper {
+        overflow: hidden;
+        height: calc(100% / 1.78);
+        width: 100%;
+        position: relative;
+    }
+    .dish-label {
+        background-color: #f6f5f1;
+        border-radius: 12px;
+        border-top-left-radius: 0;
```
Comment (DiogoGeada) [inline src/components/Dish/Dish.vue]: replace the 3 lines with border-radius: 0 12px 12px 0;

File: tests/unit/dish.test.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    test('if hiding the space of the image', () => {
+        const image = require('@/assets/images/All-American-Bacon-Cheeseburgers_exps48107_TH2379798C03_29_1b_RMS.jpg'); 
+        let wrapper = mount(Dish, {
+            propsData: { image }
+        });
+    });
+    test('if we have a title that is too long', () => {
+
+    });  
+    test('if it loads the cover image', () => {
```
Comment (DiogoGeada) [inline tests/unit/dish.test.js]: Forget this test. Jest is not very good to test images;

File: tests/unit/dish.test.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    })
+    test('if there are too many characters in description', () => {
+        const description = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse porttitor feugiat aliquam. In euismod quis tellus id fringilla. Maecenas est magna, tempus eleifend pellentesque quis, porta in dolor. Cras facilisis pellentesque diam, sed cursus nulla pellentesque et. Proin sit amet feugiat augue. Ut ornare, odio ac volutpat tincidunt, lorem lorem porttitor quam, eu aliquet velit turpis imperdiet erat. Vivamus eu lacus gravida, tempor odio nec, tincidunt dui. Vestibulum et scelerisque neque, nec sodales justo. Ut laoreet ultrices nulla, sed fermentum mi eleifend quis."
+        let wrapper = mount(Dish, {
+            propsData: { description }
+        });
+        expect(wrapper.findComponent('p').text().length).toBeLessThanOrEqual(53)
+    });
+    /*
+    test('if hiding the space of the image', () => {
```
Comment (DiogoGeada) [inline tests/unit/dish.test.js]: This test and the next need to be done.

**Review by DiogoGeada — DISMISSED** [review]

File: src/components/Dish/Dish.vue (hunk 1 above)
Comment (jespernpedersen) [reply src/components/Dish/Dish.vue]: Resolved

File: src/components/Dish/Dish.vue (hunk 2 above)
Comment (jespernpedersen) [reply src/components/Dish/Dish.vue]: Resolved

File: src/components/Dish/Dish.vue (hunk 3 above)
Comment (jespernpedersen) [reply src/components/Dish/Dish.vue]: Resolved

File: tests/unit/dish.test.js (hunk 4 above)
Comment (jespernpedersen) [reply tests/unit/dish.test.js]: Resolved

**Review by akram0171 — APPROVED** [review]

**Pull request merged**